### ISSUE TITLE
Best Practice for Transcript Retrieval

### ISSUE THREAD

**Issue by adithram** [dialpad/dialpad-python-sdk#11]
Hi! 

Apologies in advance for creating this as a Git issue - I just figured this may be the easiest way to get some technical guidance from your team...

We're currently working on an implementation using the Dialpad API (and preferably through the Python SDK) where we need to consistently/constantly retrieve all of the transcripts of all conversations of the workforce at the company. 

So far, when reading through API documentation, it appears that the best way to do this would be to:

1. Submit a `POST` request to the `/statistics` endpoint for a given date range. 
2. Poll for completion of the asynchronous statistics job via a `GET`request to the `/statistics` endpoint. 
3. Upon completion of the request, retrieve all available `call_id` values. 
4. Submit a `GET` request to the `transcript` endpoint specifying the `call_id` in the URL.

Is this the best way of performing this process? The portion of this process that seems like it could be potentially brittle is the asynchronous + polling nature of the `statistics` requests. 

We know that there is also a webhook/event driven mechanism, but are trying to avoid hosting additional infrastructure for the sake of identifying relevant `call_ids`. 

TLDR: What is the best way for us to retrieve `call_ids` for a given date range WITHOUT hosting additional infrastructure?

**Comment by dp-rufus** [comment]
Hi Adithya, I think you've identified the second best approach if you're not able to use call event subscriptions for this (which would be preferable I think)

**Comment by mstaunton** [comment]
Hi!

We were able to successfully retrieve the call_ids for the companies workforce via the approach above, which is working well. I had a follow up question regarding the return response of the [Get Call Endpoint](https://developers.dialpad.com/reference/callapi_getcallinfo).

1. When looking at the response for a call, I see there is a field `was_recorded` is set to True for some calls, but empty for others. When the field is set to True, the field `call_recording_ids` exists and has contains an ID. I tried to query the call endpoint with this id, but I am getting an error. Only some of these calls have the `transcription_content` field. Is there a way to get the transcript from the `call_recording_ids`?

**Comment by dp-rufus** [comment]
> When the field is set to True, the field call_recording_ids exists and has contains an ID. I tried to query **the call endpoint** with this id, but I am getting an error. 

Is this a typo - do you mean the `transcripts/<call_id>` endpoint?

**Comment by mstaunton** [comment]
I just tried the `transcripts/<call_id>` endpoint with a id pulled from the `call_recordings_ids` field and it returns 404. The same thing happens when making a call to `calls/<call_id>` endpoint. I am not exactly sure what the id from the `call_recordings_id` is used for. 

The reason I am using the calls endpoint is because we are also interested in some metadata about the call as well that is not returned with the transcript endpoint (ie participants etc)

**Comment by dp-rufus** [comment]
`call_recording_id` is not the same thing as `call_id`, it refers to the actual audio recording of a call, which isn't what you want here

**Comment by mstaunton** [comment]
So `was_recorded` field means there is an actual audio recording of the conversation, but does not necessarily mean that there is a transcript/`transcription_text` field for the call. Is that correct? 

Is there some configuration needed to ensure that all recorded calls are transcribed?

**Comment by dp-rufus** [comment]
`was_recorded` indicates there is an audio recording.

I'm not sure what circumstances would result in `transcript_text` being unavailable despite there being a recording. I *think* the best you can do to get a transcript is ensure VI is turned on.

Questions about transcripts are best raised through the normal support channels so they are routed to the right people.
